daemon/k8s_watcher.go

```diff
@@ -1452,6 +1455,53 @@ func (d *Daemon) deleteCiliumNetworkPolicyV2(cnp *cilium_v2.CiliumNetworkPolicy)
 	}
 }
 
+func updateRuleAnnotations(ciliumV2Store cache.Store, cnp *cilium_v2.CiliumNetworkPolicy) error {
+
+	// Get the most recent version of this CNP, as it may have been updated.
```

Comment:
@aanm this is the reasoning behind the use of the cache.Store, so that we do not update older versions of objects?